dataRetriever/factory/resolverscontainer/baseResolversContainerFactory.go

```diff
@@ -97,6 +100,10 @@ func (brcf *baseResolversContainerFactory) checkParams() error {
 	if brcf.numFullHistoryPeers <= 0 {
 		return fmt.Errorf("%w for numFullHistoryPeers", dataRetriever.ErrInvalidValue)
 	}
+	if brcf.maxNumOfValidatorInfoInResponse < minNumOfValidatorInfoInResponse {
```

Comment:
please remove this + the const defined here. The factory should not care about the params used in the instances it creates because this will just double the code logic & defined constants